### PR TITLE
Fixed some of the failing parquet_tests [databricks]

### PR DESCRIPTION
**Pull request by razajafri**
This PR contributes towards fixing #11024

### CODE REVIEW

File: integration_tests/src/main/python/parquet_test.py [hunk 1]
```diff
@@ -35,15 +35,19 @@ def read_parquet_df(data_path):
 def read_parquet_sql(data_path):
     return lambda spark : spark.sql('select * from parquet.`{}`'.format(data_path))
 
+datetimeRebaseModeInWriteKey = 'spark.sql.legacy.parquet.datetimeRebaseModeInWrite' if is_before_spark_400() else 'spark.sql.parquet.datetimeRebaseModeInWrite'
```
Comment (revans2) [inline integration_tests/src/main/python/parquet_test.py:38]: All of the non-legacy versions of these configs appear to have been added in 3.0.0. Is there a reason we are not just switching over to using them instead?

File: sql-plugin/src/main/spark400/scala/com/nvidia/spark/rapids/shims/GpuBatchScanExec.scala [hunk 2]
```diff
@@ -47,7 +47,7 @@ case class GpuBatchScanExec(
   @transient override lazy val batch: Batch = if (scan == null) null else scan.toBatch
   // TODO: unify the equal/hashCode implementation for all data source v2 query plans.
   override def equals(other: Any): Boolean = other match {
-    case other: GpuBatchScanExec =>
+    case other: BatchScanExec =>
```
Comment (revans2) [inline sql-plugin/src/main/spark400/scala/com/nvidia/spark/rapids/shims/GpuBatchScanExec.scala:50]: Why is this being changed? This is a GpuBatchScanExec. We don't want to be equal to non-GPU versions do we?

Comment (razajafri) [reply sql-plugin/src/main/spark400/scala/com/nvidia/spark/rapids/shims/GpuBatchScanExec.scala:50]: Right. While debugging I wasn't sure what was causing a failure and looking at 330 [shim](https://github.com/NVIDIA/spark-rapids/blob/7de3fc4a9b23b819be632c6cfea85a706cdafef3/sql-plugin/src/main/spark330/scala/com/nvidia/spark/rapids/shims/GpuBatchScanExec.scala#L49) I changed this and didn't change it back before submitting this PR. 

I am adding that change to this PR as well.

**Comment by razajafri** [comment]
build

**Comment by razajafri** [comment]
The failure in CI seems unrelated 
```
 - extract mortgage data *** FAILED ***

[2024-09-06T17:53:06.482Z]   0 did not equal 10000 (MortgageSparkSuite.scala:65)
```
The test only reads a CSV, sorts it and does a row count. It passes locally

**Comment by razajafri** [comment]
build

**Comment by jlowe** [comment]
> The failure in CI seems unrelated

This is tracked in #11436, test was temporarily disabled in #11451.

**Comment by razajafri** [comment]
```
We currently do not have sufficient g5.4xlarge capacity in the Availability Zone you requested (us-west-2c). Our system will be working on provisioning additional capacity. You can currently get g5.4xlarge capacity by not specifying an Availability Zone in your request or choosing us-west-2a, us-west-2b.
```

**Comment by razajafri** [comment]
build

**Pull request merged**